scripts/run_action.mjs

```diff
@@ -46,32 +35,72 @@ const resolveActionPath = async actionPath => {
     for (const extension of extensions) {
       const pathCandidate = `${path.resolve(root, actionPath)}.action.${extension}`;
 
-      if (fs.existsSync(pathCandidate)) {
+      if (existsSync(pathCandidate)) {
         return pathCandidate;
       }
     }
   }
 };
 
+/**
+ * @description promisifies the child process (for supporting legacy bash actions!)
+ */
+const spawnStream = (command, parameters) =>
+  new Promise((resolve, reject) => {
+    const childProcess = spawn(command, parameters);
+
+    childProcess.stdout.on("data", data => console.info(data.toString()));
+    childProcess.stderr.on("data", error => console.error(chalk.red(error.toString())));
+
+    childProcess.on("close", code => {
+      if (code === 0) {
+        resolve(childProcess);
+      } else {
+        reject(childProcess);
+      }
+    });
+  });
+
+/**
+ * @description This is the entrypoint into our custom task runner.
+ * It runs an `*.action.mjs` or `*.action.sh` file with the given parameters.
+ * You can find these files located within their associated directories in the project.
+ * Action files can also define:
+ *   `requirements` - actions that must be run beforehand as prerequisite.
+ *   `dependencies` - files and folders that the action takes as input. If they remain unchanged, the action can be skipped.
+ *
+ * @param {string} actionPath The truncated path to the action you wish to run (e.g. "www/build")
+ * @param {...string} parameters The flags and other parameters we want to run the action with.
+ * @returns {void}
+ */
 export async function runAction(actionPath, ...parameters) {
   const resolvedPath = await resolveActionPath(actionPath);
   if (!resolvedPath) {
     console.info("Please provide an action to run.");
     return runAction("list");
   }
 
-  const startTime = performance.now();
   console.group(chalk.yellow.bold(`▶ action(${actionPath}):`));
+  const startTime = performance.now();
 
   try {
     if (resolvedPath.endsWith("mjs")) {
-      const action = await import(resolvedPath);
+      const action = os.platform().startsWith("win")
+        ? await import(`file://${resolvedPath}`)
+        : await import(resolvedPath);
+
+      if (action.requirements?.length) {
+        for (const requiredAction of action.requirements) {
```

Comment:
We need to be able pass different parameters to different child actions. This will require a bit of a redesign. It probably means the parent action telling you what the child actions and their parameters are, given the parent parameters.